corehq/apps/export/models/new.py

```diff
@@ -316,10 +340,19 @@ def generate_schema_from_builds(domain, app_id, form_xmlns):
         :param domain: The domain that the export belongs to
         :param app_id: The app_id that the export belongs to
         :param unique_form_id: The unique identifier of the item being exported
-        :returns: Returns a ExportDataSchema instance
+        :returns: Returns a FormExportDataSchema instance
         """
-        app_build_ids = get_built_app_ids_for_app_id(domain, app_id)
-        all_xform_conf = ExportDataSchema()
+        xform_schema_id = get_latest_form_export_schema_id(domain, app_id, form_xmlns)
+        if xform_schema_id:
+            current_xform_schema = FormExportDataSchema.get(xform_schema_id)
+        else:
+            current_xform_schema = FormExportDataSchema()
```

Comment:
This section could probably be a little more succinct if `get_latest_form_export_schema_id()` returned the doc itself instead of just the id.